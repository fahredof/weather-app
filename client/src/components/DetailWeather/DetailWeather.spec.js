import React from 'react';
import {DetailWeather} from "./DetailWeather";

describe(`Details`, () => {
    const mainCity = {
        wind: `Wind 3 m/s`,
        overcast: `overcast clouds`,
        pressure: `Pressure 1011 hpa`,
        humidity: `Humidity 66 %`,
        coordinates: `Coordinates [54.73, 55.95]`,
    };
    const mainCityError = {
        wind: ``,
        overcast: ``,
        pressure: ``,
        humidity: ``,
        coordinates: ``,
    };
    const props = {
        mainCity
    };

    describe(`отображение города`, () => {
        const defCity = shallow(<DetailWeather {...props}/>);
        it(`отображается корректно`, () => expect(shallowToJson(defCity)).toMatchSnapshot());
    });

    describe('ошибка при получении города', () => {
        const nextProps = {
            ...props,
            mainCity: mainCityError
        };
        const defCity = shallow(<DetailWeather {...nextProps} />);
        it('отображается корректно', () => expect(shallowToJson(defCity)).toMatchSnapshot());
    });
});