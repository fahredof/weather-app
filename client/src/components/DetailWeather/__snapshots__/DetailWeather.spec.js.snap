// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details отображение города отображается корректно 1`] = `
<div
  className="details"
>
  <Details
    mainCity={
      Object {
        "coordinates": "Coordinates [54.73, 55.95]",
        "humidity": "Humidity 66 %",
        "overcast": "overcast clouds",
        "pressure": "Pressure 1011 hpa",
        "wind": "Wind 3 m/s",
      }
    }
  />
</div>
`;

exports[`Details ошибка при получении города отображается корректно 1`] = `
<div
  className="details"
>
  <Spinner />
</div>
`;
